### PR TITLE
Tests for NXOS

### PR DESCRIPTION
**Pull request by mirceaulinic**
+ few other small issues fixed

### CODE REVIEW

**Comment by mirceaulinic** [comment]
@dbarrosop This is the PR I was talking about yesterday. Main purpose is to include unit tests for NXOS and also to fix a couple of small bugs that are blocking #160 and #183

File: napalm/iosxr.py [hunk 1]
```diff
@@ -916,13 +915,12 @@ def get_arp_table(self):
 
         result_tree = ET.fromstring(self.device.make_rpc_call(rpc_command))
 
-        for arp_entry in result_tree.findall('.//ResolutionHistoryDynamic/Entry/Entry'):
```
Comment (mirceaulinic) [inline napalm/iosxr.py]: Refactored get_arp_table for IOS-XR

File: napalm/junos.py [hunk 2]
```diff
@@ -336,16 +336,26 @@ def get_lldp_neighbors_detail(self, interface = ''):
 
         lldp_table = junos_views.junos_lldp_neighbors_detail_table(self.device)
         lldp_table.get()
-
```
Comment (mirceaulinic) [inline napalm/junos.py]: Tricky detailed view of LLDP neighbors on JunOS: https://github.com/Juniper/py-junos-eznc/issues/474

File: test/unit/junos/mock_data/get-lldp-neighbors-information.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <lldp-neighbors-information style="brief">
   <lldp-neighbor-information>
-    <lldp-local-interface>xe-0/1/2.0</lldp-local-interface>
```
Comment (mirceaulinic) [inline test/unit/junos/mock_data/get-lldp-neighbors-information.txt]: On some devices ```lldp-local-interface``` is renamed to ```lldp-local-port-id```. Must investigate that later.

**Comment by dbarrosop** [comment]
Great job!

**Pull request merged**